config/src/config.rs

```diff
@@ -128,8 +129,17 @@ impl GlobalConfig {
 				// Put the struct back together, because the config
 				// file was flattened a bit
 				gc.server.stratum_mining_config = gc.mining_server.clone();
+				if Path::new(&gc.server.db_root).is_relative() {
```

Comment:
I think the `config` crate should just worry about finding, reading and parsing the config. Massaging the diverse options seems better under `src/bin`. Maybe add a `config.rs` helper there to encapsulate all config manipulations after reading?

Reply:
logic moved to src/bin